src/simplewallet/simplewallet.cpp

```diff
@@ -465,6 +476,565 @@ bool simple_wallet::print_fee_info(const std::vector<std::string> &args/* = std:
+  if(orig_pwd_container == boost::none)
+  {
+    fail_msg_writer() << tr("Your password is incorrect.");
+    return true;
+  }
+
+  std::string multisig_info = m_wallet->get_multisig_info();
+  success_msg_writer() << multisig_info;
+  success_msg_writer() << tr("Send this multisig info to all other participants, then use make_multisig <threshold> <info1> [<info2>...] with others' multisig info");
+  success_msg_writer() << tr("This includes the PRIVATE view key, so needs to be disclosed only to that multisig wallet's participants ");
+  return true;
+}
+
+bool simple_wallet::make_multisig(const std::vector<std::string> &args)
+{
+  if (m_wallet->multisig())
+  {
+    fail_msg_writer() << tr("This wallet is already multisig");
+    return true;
+  }
+  if (m_wallet->watch_only())
+  {
+    fail_msg_writer() << tr("wallet is watch-only and cannot be made multisig");
+    return true;
+  }
+
+  if(m_wallet->balance())
+  {
+    fail_msg_writer() << tr("This wallet has a balance, making it multisig would lose it");
+    return true;
+  }
+
+  const auto orig_pwd_container = get_and_verify_password();
+  if(orig_pwd_container == boost::none)
+  {
+    fail_msg_writer() << tr("Your original password was incorrect.");
+    return true;
+  }
+
+  if (args.size() < 2)
+  {
+    fail_msg_writer() << tr("usage: make_multisig <threshold> <multisiginfo1> [<multisiginfo2>...]");
+    return true;
+  }
+
+  // parse threshold
+  uint32_t threshold;
+  if (!string_tools::get_xtype_from_string(threshold, args[0]))
+  {
+    fail_msg_writer() << tr("Invalid threshold");
+    return true;
+  }
+
+  // parse all multisig info
+  std::vector<crypto::secret_key> secret_keys(args.size() - 1);
+  std::vector<crypto::public_key> public_keys(args.size() - 1);
+  for (size_t i = 1; i < args.size(); ++i)
+  {
+    if (!m_wallet->verify_multisig_info(args[i], secret_keys[i - 1], public_keys[i - 1]))
+    {
+      fail_msg_writer() << tr("Bad multisig info: ") << args[i];
+      return true;
+    }
+  }
+
+  // remove duplicates
+  for (size_t i = 0; i < secret_keys.size(); ++i)
+  {
+    for (size_t j = i + 1; j < secret_keys.size(); ++j)
+    {
+      if (rct::sk2rct(secret_keys[i]) == rct::sk2rct(secret_keys[j]))
+      {
+        message_writer() << tr("Duplicate key found, ignoring");
+        secret_keys[j] = secret_keys.back();
+        public_keys[j] = public_keys.back();
+        secret_keys.pop_back();
+        public_keys.pop_back();
+        --j;
+      }
+    }
+  }
+
+  // people may include their own, weed it out
+  for (size_t i = 0; i < secret_keys.size(); ++i)
+  {
+    if (rct::sk2rct(secret_keys[i]) == rct::sk2rct(m_wallet->get_account().get_keys().m_view_secret_key))
+    {
+      message_writer() << tr("Local key is present, ignoring");
+      secret_keys[i] = secret_keys.back();
+      public_keys[i] = public_keys.back();
+      secret_keys.pop_back();
+      public_keys.pop_back();
+      --i;
```

Comment:
Shouldn't it be possible to simply `break` here, since any duplicates are guaranteed to have been eliminated already?

Reply:
I think yes, technically. This is more resistant to code reorg though.